grid.v3/codec/registry.go

```diff
@@ -1,61 +1,90 @@
 package codec
 
 import (
+	"errors"
 	"reflect"
 	"sync"
+
+	"github.com/gogo/protobuf/proto"
 )
 
-var registry *codecRegistry = &codecRegistry{reg: map[string]Codec{}} //shared registry
+var (
+	ErrNonProtoMessage         = errors.New("codec: non proto message")
+	ErrUnregisteredMessageType = errors.New("codec: unregistered message type")
+)
 
-type CodecRegistry interface {
-	Register(v interface{}, codec Codec)
-	GetCodec(v interface{}) (Codec, bool)
-	GetCodecName(name string) (Codec, bool)
-}
+var (
+	mu       = &sync.RWMutex{}
+	registry = map[string]interface{}{}
+)
 
-func Registry() CodecRegistry {
-	return registry
-}
+func Register(v interface{}) error {
```

Comment:
Yea lets keep the package function.   In this case its really not a big deal, just Yak Shaving.